dspace-server-webapp/src/test/java/org/dspace/app/rest/authorization/EPersonRegistrationFeatureIT.java

```diff
@@ -0,0 +1,118 @@
+import org.dspace.app.rest.converter.ConverterService;
+import org.dspace.app.rest.converter.SiteConverter;
+import org.dspace.app.rest.model.SiteRest;
+import org.dspace.app.rest.projection.Projection;
+import org.dspace.app.rest.test.AbstractControllerIntegrationTest;
+import org.dspace.app.rest.utils.Utils;
+import org.dspace.content.Site;
+import org.dspace.content.service.SiteService;
+import org.dspace.services.ConfigurationService;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class EPersonRegistrationFeatureIT extends AbstractControllerIntegrationTest {
+
+    @Autowired
+    private AuthorizationFeatureService authorizationFeatureService;
+
+    @Autowired
+    private ConverterService converterService;
+
+    @Autowired
+    private ConfigurationService configurationService;
+
+    @Autowired
+    private SiteService siteService;
+
+    @Autowired
+    private SiteConverter siteConverter;
+
+    @Autowired
+    private Utils utils;
+
+    private AuthorizationFeature epersonRegistrationFeature;
+
+    public static final String[] SHIB_ONLY = {"org.dspace.authenticate.ShibAuthentication"};
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        epersonRegistrationFeature = authorizationFeatureService.find(EPersonRegistrationFeature.NAME);
+    }
+
+    @Test
+    public void userRegistrationEnabledSuccessTest() throws Exception {
+
+        Site site = siteService.findSite(context);
+        SiteRest SiteRest = siteConverter.convert(site, Projection.DEFAULT);
+        String siteUri = utils.linkToSingleResource(SiteRest, "self").getHref();
+
+
+        getClient().perform(get("/api/authz/authorizations/search/objectAndFeature")
+                                          .param("uri", siteUri)
+                                          .param("feature", epersonRegistrationFeature.getName()))
+                             .andExpect(status().isOk());
+    }
+
+    @Test
+    public void userRegistrationDisabledUnAuthorizedTest() throws Exception {
+
+        Site site = siteService.findSite(context);
+        SiteRest SiteRest = siteConverter.convert(site, Projection.DEFAULT);
+        String siteUri = utils.linkToSingleResource(SiteRest, "self").getHref();
+
+        context.turnOffAuthorisationSystem();
+        configurationService.setProperty("user.registration", false);
+        context.restoreAuthSystemState();
+
+        getClient().perform(get("/api/authz/authorizations/search/objectAndFeature")
+                                          .param("uri", siteUri)
+                                          .param("feature", epersonRegistrationFeature.getName()))
+                             .andExpect(status().isNoContent());
+
+        context.turnOffAuthorisationSystem();
+        configurationService.setProperty("user.registration", true);
+        context.restoreAuthSystemState();
```

Comment:
Same as above, the two context lines aren't doing anything.  As a sidenote too, the configurationService reloads between test classes, so you don't need to reset the configuration properly here if you don't want to.  In other words, these three lines could be removed

Reply:
Minor update / sidenote:  I noticed others making this same honest mistake with regards to changing configuration properties in tests. So, I added this to our Code Testing Guide: https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide#CodeTestingGuide-Changingconfigurationpropertiesintests

So, just to clarify for the future.  Changing config values doesn't require messing with the context.  And you don't need to change them back to defaults (it'll happen automatically).  But, both of those notes are now in the testing guide!

Reply:
This has been adjusted